src/devices/cpu/tlcs900/tmp94c241.h

```diff
@@ -0,0 +1,419 @@
+#define TMP94C241_INTET01  4
+#define TMP94C241_INTET23  5
+#define TMP94C241_INTET45  6
+#define TMP94C241_INTET67  7
+#define TMP94C241_INTET89  8
+#define TMP94C241_INTETAB  9
+#define TMP94C241_INTES0   10
+#define TMP94C241_INTES1   11
+#define TMP94C241_INTETC01 12
+#define TMP94C241_INTETC23 13
+#define TMP94C241_INTETC45 14
+#define TMP94C241_INTETC67 15
+#define TMP94C241_INTE0AD  16
+#define TMP94C241_INTNMWDT 17
+
+// Flip-Flop operations:
+#define FLIPFLOP_INVERT   0b00
+#define FLIPFLOP_SET      0b01
+#define FLIPFLOP_CLEAR    0b10
+#define FLIPFLOP_DONTCARE 0b11
+
+//**************************************************************************
+//  TYPE DEFINITIONS
+//**************************************************************************
+
+// ======================> tmp94c241_device
+
+class tmp94c241_device : public tlcs900h_device
+{
+public:
+	// device type constructor
+	tmp94c241_device(const machine_config &mconfig, const char *tag, device_t *owner, u32 clock);
+
+	// configuration helpers
+	auto port0_read()  { return m_port0_read.bind(); }
+	auto port0_write() { return m_port0_write.bind(); }
+	auto port1_read()  { return m_port1_read.bind(); }
+	auto port1_write() { return m_port1_write.bind(); }
+	auto port2_read()  { return m_port2_read.bind(); }
+	auto port2_write() { return m_port2_write.bind(); }
+	auto port3_read()  { return m_port3_read.bind(); }
+	auto port3_write() { return m_port3_write.bind(); }
+	auto port4_read()  { return m_port4_read.bind(); }
+	auto port4_write() { return m_port4_write.bind(); }
+	auto port5_read()  { return m_port5_read.bind(); }
+	auto port5_write() { return m_port5_write.bind(); }
+	auto port6_read()  { return m_port6_read.bind(); }
+	auto port6_write() { return m_port6_write.bind(); }
+	auto port7_read()  { return m_port7_read.bind(); }
+	auto port7_write() { return m_port7_write.bind(); }
+	auto port8_read()  { return m_port8_read.bind(); }
+	auto port8_write() { return m_port8_write.bind(); }
+	auto porta_read()  { return m_porta_read.bind(); }
+	auto porta_write() { return m_porta_write.bind(); }
+	auto portb_read()  { return m_portb_read.bind(); }
+	auto portb_write() { return m_portb_write.bind(); }
+	auto portc_read()  { return m_portc_read.bind(); }
+	auto portc_write() { return m_portc_write.bind(); }
+	auto portd_read()  { return m_portd_read.bind(); }
+	auto portd_write() { return m_portd_write.bind(); }
+	auto porte_read()  { return m_porte_read.bind(); }
+	auto porte_write() { return m_porte_write.bind(); }
+	auto portf_read()  { return m_portf_read.bind(); }
+	auto portf_write() { return m_portf_write.bind(); }
+	auto portg_read()  { return m_portg_read.bind(); }
+	auto porth_read()  { return m_porth_read.bind(); }
+	auto porth_write() { return m_porth_write.bind(); }
+	auto portz_read()  { return m_portz_read.bind(); }
+	auto portz_write() { return m_portz_write.bind(); }
+
+protected:
+	// device-level overrides
+	virtual void device_config_complete() override;
```

Comment:
Minor thing, but please don’t copy/paste this comment – it doesn’t even make sense.  Use “device_t implementation” or similar.